### PR TITLE
[REVIEW] Added base for Dagger 2 injections

### PR DESCRIPTION
**Pull request by jaggernod**
Added base modules for injecting ViewModels, Fetchers and network related items

### CODE REVIEW

File: app/src/main/java/com/tehmou/rxbookapp/data/stores/StoreModule.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import dagger.Provides;
+
+/**
+ * Created by Pawel Polanski on 5/16/15.
+ */
+@Module
+public final class StoreModule {
+
+    @Provides
+    @Singleton
```
Comment (tehmou) [inline app/src/main/java/com/tehmou/rxbookapp/data/stores/StoreModule.java]: Is there a reason only UserSettingsStore is singleton?

Comment (jaggernod) [inline app/src/main/java/com/tehmou/rxbookapp/data/stores/StoreModule.java]: Is UserSettingsStore bound to single user or all users?

Comment (tehmou) [inline app/src/main/java/com/tehmou/rxbookapp/data/stores/StoreModule.java]: I mean the other stores are not singletons, should they be as well? The UserStore is indeed a global store that is not contextualized to any actual user.

Comment (jaggernod) [inline app/src/main/java/com/tehmou/rxbookapp/data/stores/StoreModule.java]: You are right, in that case, all the stores should be singletons.

Comment (jaggernod) [inline app/src/main/java/com/tehmou/rxbookapp/data/stores/StoreModule.java]: DONE

**Pull request merged**